### PR TITLE
Fix handling of saving documents after modifications, add tests

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

File: pyArango/document.py [hunk 1]
```diff
@@ -373,21 +386,49 @@ def getPatches(self) :
...
+    def __len__(self):
+        if not self._store:
+            return 0
+
+        return self._store.__len__()
+
+    def __dict__(self):
+        if not self._store:
+            return []
+        return dict(self._store)
```
Comment (dilipvamsi) [inline pyArango/document.py]: The function should return a dictionary.
```python
    def __dict__(self):
        if not self._store:
            return {}
        return dict(self._store)
```

Comment (dothebart) [reply pyArango/document.py]: thanks, fixed.

**Comment by tariqdaouda** [comment]
Could you please resend this to the dev branch?

**Comment by tariqdaouda** [comment]
Thank you @dothebart could you please send this to the dev branch?

**Pull request merged**